### PR TITLE
Adds class trim method to VideoSettings, uses it for title and desc

### PR DESCRIPTION
**Pull request by cjlarsen**
<b>Ticket</b>
https://vimean.atlassian.net/browse/VIM-3848

<b>Ticket Summary</b>
"Something strange occurred" alert when you upload video with just spaces in title

<b>Implementation Summary</b>

Trim whitespace from title before starting upload
How To Test
1. open the app 
2. upload your video 
3. tap on a video and set your title with only spaces 
4. tap [upload]
5. Title is now "Untitled"

### CODE REVIEW

**Comment by cjlarsen** [comment]
@ghking Ready to review, thanks

File: VimeoUpload/Upload/Model/VideoSettings.swift [hunk 1]
```diff
@@ -29,18 +29,49 @@ import Foundation
 class VideoSettings: NSObject
 {
     var title: String?
+    {
+        didSet
+        {
+            if let title = self.title
+            {
+                self.title = trim(title)
+            }
```
Comment (ghking) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: Let's make the argument to trim() optional so we don't have to check `if let title = self.title`

File: VimeoUpload/Upload/Model/VideoSettings.swift [hunk 2]
```diff
@@ -29,18 +29,49 @@ import Foundation
...
     var users: [String]? // List of uris of users who can view this video
     var password: String?
 
     init(title: String?, description: String?, privacy: String?, users: [String]?, password: String?)
     {
-        self.title = title
-        self.desc = description
         self.privacy = privacy
         self.users = users
         self.password = password
```
Comment (ghking) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: Should we move these underneath the super call as well?

File: VimeoUpload/Upload/Model/VideoSettings.swift [hunk 3]
```diff
@@ -96,4 +127,11 @@ class VideoSettings: NSObject
         aCoder.encodeObject(self.users, forKey: "users")
         aCoder.encodeObject(self.password, forKey: "password")
     }
+    
+    // MARK : String Methods
+    
+    func trim(text: String) -> String
+    {
+        return text.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet())
+    }
```
Comment (ghking) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: Yeah I totally forgot that trim() from our own extension inside the app code. Nice catch

**Comment by ghking** [comment]
DWR

**Comment by cjlarsen** [comment]
@ghking ok updated

File: VimeoUpload/Upload/Model/VideoSettings.swift [hunk 4]
```diff
@@ -96,4 +112,16 @@ class VideoSettings: NSObject
...
+    // MARK : String Methods
+    
+    func trim(text: String?) -> String
+    {
+        if let constText = text
+        {
+            return constText.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet())
+        }
+        
+        return ""
```
Comment (ghking) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: Let's make the return type optional as well (I think it'd be weird if trimming nil returns something other than nil). Then we can condense this into a single line. After that let's merge this in!

File: VimeoUpload/Upload/Model/VideoSettings.swift [hunk 5]
```diff
@@ -96,4 +112,16 @@ class VideoSettings: NSObject
...
+    // MARK : String Methods
+    
+    func trim(text: String?) -> String?
+    {
+        if let constText = text
+        {
+            return constText.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet())
+        }
+        
+        return nil
```
Comment (ghking) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: This can just be:

`return text?.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet())'

After that we should be good to merge!

Comment (cjlarsen) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: Sounds good, done, merging

File: VimeoUpload/Upload/Model/VideoSettings.swift [hunk 6]
```diff
@@ -29,18 +29,34 @@ import Foundation
 class VideoSettings: NSObject
 {
     var title: String?
+    {
+        didSet
+        {
+            self.title = trim(self.title)
```
Comment (alfiehanssen) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: @cjlarsen A couple comments to address, 

Whenever we invoke an instance method we prefix it with `self.` (we should add that here)

File: VimeoUpload/Upload/Model/VideoSettings.swift [hunk 7]
```diff
@@ -29,18 +29,34 @@ import Foundation
...
+        }
+    }
+    
     var desc: String?
+    {
+        didSet
+        {
+            self.desc = trim(self.desc)
```
Comment (alfiehanssen) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: Same here

File: VimeoUpload/Upload/Model/VideoSettings.swift [hunk 8]
```diff
@@ -29,18 +29,34 @@ import Foundation
...
     {
-        self.title = title
-        self.desc = description
+        super.init()
+        
         self.privacy = privacy
         self.users = users
         self.password = password
+        self.title = self.trim(title)
+        self.desc = self.trim(description)
```
Comment (alfiehanssen) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: This seems redundant. We're trimming here before we invoke the setter. And then above in the setting we're trimming as well. Should probably not trim here in init and just let the setter handle it all.

**Comment by alfiehanssen** [comment]
Cc @ghking, added a few more comments

File: VimeoUpload/Upload/Model/VideoSettings.swift (hunk 8 above)
Comment (cjlarsen) [inline VimeoUpload/Upload/Model/VideoSettings.swift]: @alfiehanssen Hey sorry for taking so long to get back on this.  Unfortunately didSet is never called from init in swift.  

from the apple docs: 

> The willSet and didSet observers of superclass properties are called when a property is set in a subclass initializer, after the superclass initializer has been called. They are not called while a class is setting its own properties, before the superclass initializer has been called.

apparently wrapping it in a closure calls didSet.  Going to submit another pull request with that and the other updates.
